From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 28 Apr 2025 21:47:22 +0200
Subject: Remove '#include <linux/prctl.h>'

To fix this error:

    In file included from mimalloc-2.2.3/src/prim/unix/prim.c:36,
                     from mimalloc-2.2.3/src/prim/prim.c:25:
    /usr/include/sys/prctl.h:88:8: error: redefinition of 'struct prctl_mm_map'
       88 | struct prctl_mm_map {
          |        ^~~~~~~~~~~~
    In file included from mimalloc-2.2.3/src/prim/unix/prim.c:34:
    /usr/include/linux/prctl.h:134:8: note: originally defined here
      134 | struct prctl_mm_map {
          |        ^~~~~~~~~~~~

Upstream-Issue: https://github.com/microsoft/mimalloc/issues/1056

--- a/src/prim/unix/prim.c
+++ b/src/prim/unix/prim.c
@@ -31,7 +31,6 @@
 
 #if defined(__linux__)
   #include <features.h>
-  #include <linux/prctl.h>  // PR_SET_VMA
   //#if defined(MI_NO_THP)
   #include <sys/prctl.h>    // THP disable
   //#endif
